api/bots_api/bots_test_lib.py

```diff
@@ -91,18 +91,33 @@ def mock_http_conversation(self, test_name):
             os.path.abspath(__file__)), '..', 'bots', self.bot_name, 'fixtures'))
         http_request_path = os.path.join(base_path, '{}_request.json'.format(test_name))
         http_response_path = os.path.join(base_path, '{}_response.json'.format(test_name))
+        http_headers_path = os.path.join(base_path, '{}_headers.json'.format(test_name))
         with open(http_request_path, 'r') as http_request_file, \
-                open(http_response_path, 'r') as http_response_file:
+                open(http_response_path, 'r') as http_response_file, \
+                open(http_headers_path, 'r') as http_headers_file:
             http_request = json.load(http_request_file)
             http_response = json.load(http_response_file)
+            http_headers = json.load(http_headers_file)
             with patch('requests.get') as mock_get:
                 mock_result = mock.MagicMock()
                 mock_result.json.return_value = http_response
                 mock_result.ok.return_value = True
+                mock_result.status_code = 200
+                try:
+                    mock_result.status_code = http_headers['response']['status']
+                except KeyError:
+                    pass
+                try:
+                    mock_result.ok.return_value = http_headers['response']['ok']
+                except KeyError:
+                    pass
```

Comment:
Changes to the mock http part are in a separate commit already. I was thinking of having just one file per test as I mention in my comment below.